modules/standard/Sets.chpl

```diff
@@ -0,0 +1,395 @@
+      common.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` intersect.
+    */
+    proc const isIntersecting(const ref other: set(eltType, ?)): bool {
+      return false;
+    }
+
+    /*
+      Remove the item from this set with a value equal to `x`.
+
+      .. warning::
+
+        Removing an element from this set may invalidate existing references
+        to the elements contained in this set.
+    
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If the list contains no such element.
+    */
+    proc remove(const ref x: eltType): eltType throws {
+      return;
+    }
+
+    /*
+      Remove the item `x` if it is present in this set, but do nothing if it
+      is not.
+
+      .. warning::
+
+        Discarding an element from this set may invalidate existing references
+        to the elements contained in this set.
+
+      :arg x: The element to discard.
+    */
+    proc discard(const ref x: eltType) {
+      return;
+    }
+
+    /*
+      Clear the contents of this set.
+
+      .. warning::
+
+        Clearing the contents of this set will invalidate all existing
+        references to the elements contained in this set.
+    */
+    proc clear() {
+      return;
+    }
+
+    /*
+      Iterate over the elements of this set.
+
+      :yields: A reference to one of the elements contained in this set.
+    */
+    iter these() ref {
+      return nil;
+    }
+
+    /*
+      Write the contents of this set to a channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc const writeThis(ch: channel) {
+      return;
+    }
+
+    /*
+      Returns `true` if this set contains zero elements.
+
+      :return: `true` if this set is empty.
+      :rtype: `bool`
+    */
+    inline proc const isEmpty(): bool {
+      return false;
+    }
+
+    /*
+      The current number of elements contained in this set.
+    */
+    inline proc const size {
+      return 0;
+    }
+
+    /*
+      Returns a new DefaultRectangular array containing a copy of each of the
+      elements contained in this set. The returned array is sorted according
+      to the default sort order of its elements.
+    */
+    proc const toArray(): [] eltType {
+      return nil;
+    }
+
+  } // End record "set".
+
+  /*
+    Clear the contents of this set, then iterate through the contents of `rhs`
+    and add a copy of each element to this set.
+
+    .. warning::
+
+      This will invalidate any references to elements previously contained in
+      `lhs`.
+
+    :arg lhs: The set to assign to.
+    :arg rhs: The set to assign from. 
+  */
+  proc =(ref lhs: set(?t, ?), const ref rhs: set(t, ?)) {
+    return;
+  }
+
+  /*
+    Return a new set that contains the union of two sets.
+
+    :arg a: A set to take the difference of.
+    :arg b: A set to take the difference of.
+
+    :return: A new set containing the difference between `a` and `b`.
```

Comment:
That's a funny thing for `proc |()` to return, and for its args to be for.

Reply:
It seems to me that `|` is the natural operator to overload with the semantics of "union", do you have an alternative in mind?

Reply:
I think @cassella was just pointing out that the chpldoc string uses "difference" in several places in the arg/return descriptions

Reply:
Ha ha!

Thanks for the catch @cassella, I'll be honest I definitely copy/pasted the description from the difference docstring and forgot to change it.